webapp/src/shared/hooks/UseBlock.tsx

```diff
@@ -1,24 +1,69 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useCallback, useEffect, useReducer } from 'react';
 import { ResponseError, UserResponseDto } from '../generated';
 import { usersApi } from '../services/ApiService';
 
+export type BlockRelation = {
+  isUserBlocked: boolean;
+  amIBlocked: boolean;
+};
+
+type BlockAction =
+  | { type: 'FILL_BLOCK_RELATION'; isUserBlocked: boolean; amIBlocked: boolean }
+  | { type: 'BLOCK_USER' }
+  | { type: 'UNBLOCK_USER' };
+
+const blockReducerCallback = (
+  state: BlockRelation | null,
+  action: BlockAction,
+): BlockRelation | null => {
+  const { type } = action;
+
+  if (type === 'FILL_BLOCK_RELATION') {
+    return {
+      isUserBlocked: action.isUserBlocked,
+      amIBlocked: action.amIBlocked,
+    };
+  }
+
+  if (!state) {
+    return null;
+  }
+
+  if (type === 'BLOCK_USER') {
+    return {
+      ...state,
+      isUserBlocked: true,
+    };
+  } else if (type === 'UNBLOCK_USER') {
+    return {
+      ...state,
+      isUserBlocked: false,
+    };
+  }
```

Comment:
This may be a bit cumbersome, but good enough for me. The idea is treat `null` as a special case. Only valid to add data if `null` when filling all the state, otherwise it means we don't have a valid blockRelation